### PR TITLE
Add 2-phase build

### PR DESCRIPTION
**Pull request by damemi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,13 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
 WORKDIR /go/src/github.com/openshift/crd-schema-gen
 COPY . .
+ENV GO_PACKAGE github.com/openshift/crd-schema-gen
 RUN go build -o crd-schema-gen cmd/crd-schema-gen/main.go
+
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+WORKDIR /go/src/github.com/openshift/crd-schema-gen
+COPY --from=builder /go/src /go/src
+COPY --from=builder /usr/local/go /usr/local/go
```
Comment (damemi) [inline Dockerfile]: I'm not sure if this is the right fix, but it does fix that `Unsupported kind` error I was getting before with the second stage

@tnozicka @sttts could you review and then we can move along with https://github.com/openshift/release/pull/3428?

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
 RUN go build -o crd-schema-gen cmd/crd-schema-gen/main.go
-ENV APIS_DIR=vendor/github.com/openshift/api
-ENTRYPOINT ./crd-schema-gen --apis-dir $APIS_DIR
\ No newline at end of file
+
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+WORKDIR /go/src/generator
+COPY --from=builder /go/src/github.com/openshift/crd-schema-gen/crd-schema-gen /
+COPY --from=builder /go/src/github.com/openshift/crd-schema-gen/vendor/k8s.io/apimachinery /go/src/generator/vendor/k8s.io/apimachinery
+COPY --from=builder /go/src/github.com/openshift/crd-schema-gen/vendor/k8s.io/api /go/src/generator/vendor/k8s.io/api
```
Comment (damemi) [inline Dockerfile]: For some reason the crd-schema-gen binary needs these two dependencies to actually generate any output -- even though I'm mounting the `openshift/api` and `manifests` directories separately. @sttts could this also be related to the generator's nature as a parser, or possibly the transitive dependencies of the openshift API types that depend on the k8s api?

Comment (sttts) [reply Dockerfile]: how does the invocation of the container look like exactly?

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
-ENTRYPOINT ./crd-schema-gen --apis-dir $APIS_DIR
\ No newline at end of file
+
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+WORKDIR /go/src/generator
+COPY --from=builder /go/src/github.com/openshift/crd-schema-gen/crd-schema-gen /
+COPY --from=builder /usr/local/go /usr/local/go
+ENV GOPATH=/go
+ENV APIS_DIR=openshift/api
+ENTRYPOINT /crd-schema-gen --apis-dir $APIS_DIR
```
Comment (sttts) [inline Dockerfile]: do we really want to hard-code this? I would expect this in the Makefile of consumers.

Comment (sttts) [reply Dockerfile]: this blocks merging

Comment (damemi) [reply Dockerfile]: updated this to use `ENTRYPOINT ["/crd-schema-gen"]` and the flag can be passed in with the run command

File: README.md [hunk 4]
```diff
@@ -15,9 +15,10 @@ make images
 Then from your operator's repository run:
 
 ```
-docker run -v $(pwd)/manifests:/go/src/github.com/openshift/crd-schema-gen/manifests:Z \
-  -v $(pwd)/vendor/github.com/openshift/api:/go/src/github.com/openshift/crd-schema-gen/openshift/api:Z \
-  -e APIS_DIR="openshift/api" openshift/origin-crd-schema-gen
+docker run -v $(pwd)/manifests:/go/src/crd-schema-gen/manifests:Z \
+  -v $(pwd)/vendor:/go/src/crd-schema-gen/vendor:Z \
+  openshift/origin-crd-schema-gen \
+  --apis-dir vendor/github.com/openshift/api
```
Comment (sttts) [inline README.md]: why so complicated? why not:
```bash
PKG=${PWD#${GOPATH}/}
docker run  -v ${PWD}:/go/${PKG} -w /go/${PKG} openshift/origin-crd-schema-gen --apis-dir ....
```
Mounting into a fake dir is strange.

Comment (damemi) [reply README.md]: That is much simpler, didn't know I could set the workdir with a flag. Updated

**Pull request merged**